modules/evm/src/runner/native.rs

```diff
@@ -76,12 +72,16 @@ impl<T: Trait> RunnerT<T> for Runner<T> {
 				call_info
 			);
 
-			match reason {
-				ExitReason::Succeed(_) => TransactionOutcome::Commit(Ok(call_info)),
-				ExitReason::Revert(_) => TransactionOutcome::Rollback(Ok(call_info)),
-				ExitReason::Error(_) => TransactionOutcome::Rollback(Ok(call_info)),
-				ExitReason::Fatal(_) => TransactionOutcome::Rollback(Ok(call_info)),
+			if !reason.is_succeed() {
+				return TransactionOutcome::Rollback(Ok(call_info));
 			}
+
+			if let Err(e) = substate.transfer(Transfer { source, target, value }) {
```

Comment:
why do we need this? `substate.execute` doesn't do transfer? but shouldn't we transfer first?

Reply:
@xlc not it doesn't do transfer. I'm thinking about moving transfer inside `substrate.execute` to reduce complexity